### PR TITLE
Handle deep dependencies in yarn lock creation

### PR DESCRIPTION
**Pull request by AlonMiz**
when having child dependency of a dependency, it doesn't add that to the yarn lock
example
```
"other-dep@1.0.0":
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/in-w1-dev-dep-1.tgz#123123"
  integrity sha512-shashashashashashashhsa==
  dependencies:
    "deep-dep" "3.0.0"
```
`deep-deep` will not be contained in the final yarn lock. creating an ambiguous installation outcome.

what's in:
1. creation of yarn lock will add all deep dependencies recursively
2. handle `yarn workspace` command in new version (parse string differently)
3. add tests
4. refactor createLockFile to different file

### CODE REVIEW

File: src/create-yarn-lock.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const list = [];
+    const recursive = (dependencies = {}) => {
+      Object.entries(dependencies).forEach(([name, version]) => {
+        const depName = `${name}@${version}`;
+
+        if (!projectWorkspaces[name] && !list.includes(depName)) {
+          list.push(depName);
+          if (oldFile[depName] && oldFile[depName].dependencies) {
+            recursive(oldFile[depName].dependencies);
+          }
```
Comment (AlonMiz) [inline src/create-yarn-lock.js:25]: This is the main code change
@Madvinking

File: src/params.js [hunk 2]
```diff
@@ -57,7 +57,9 @@ const rootDir = getWorkspacesRoot(projectRoot);
 
 const rootPacakgeJson = JSON.parse(fs.readFileSync(path.join(rootDir, 'package.json'), 'utf-8'));
 
-const projectWorkspaces = JSON.parse(execSync('yarn workspaces --silent info', { cwd: rootDir }).toString());
+const projectWorkspaces = JSON.parse(
+  execSync('yarn workspaces --silent info', { cwd: rootDir, encoding: 'utf8' }).toString().replace(`[2K[1G`, ''),
```
Comment (Madvinking) [inline src/params.js:61]: ?

Comment (AlonMiz) [reply src/params.js:61]: yea. in different versions of yarn there's a different outputs... i use 1.22.10 and it generates a different json output...

**Review by Madvinking — APPROVED** [review]

**Comment by AlonMiz** [comment]
@Madvinking can you merge and publish?

**Pull request merged**